File: src/infrastructure/services/uuid-id-generator.ts
// src/infrastructure/services/uuid-id-generator.ts

import { Effect, Layer } from "effect"
import { v4 as uuidv4 } from "uuid"
import { MakeProductId, MakeVariantId } from "../../domain/pilot"
import { IdGenerator } from "../../ports/driven"

// ============================================
// UUID ID GENERATOR
// ============================================

export const UuidIdGeneratorLive = Layer.succeed(
  IdGenerator,
  {
    generateProductId: () => Effect.succeed(MakeProductId(uuidv4())),
    generateVariantId: () => Effect.succeed(MakeVariantId(uuidv4()))
  }
)

File: apps/server-java/src/main/java/com/maisonamane/infrastructure/service/UuidIdGenerator.java
package com.maisonamane.infrastructure.service;

import com.maisonamane.domain.pilot.valueobject.ProductId;
import com.maisonamane.domain.shared.CorrelationId;
import org.springframework.stereotype.Component;
import java.util.UUID;

/**
 * UUID-based ID generator.
 * Maps Effect-TS UuidIdGeneratorLive to Spring @Component.
 */
@Component
public class UuidIdGenerator implements com.maisonamane.port.driven.service.IdGenerator {

    @Override
    public ProductId generateProductId() {
        return ProductId.of(UUID.randomUUID().toString());
    }

    @Override
    public CorrelationId generateCorrelationId() {
        return CorrelationId.generate();
    }
}
